### PR TITLE
Datasource Objects and Libraries

### PR DESCRIPTION
**Pull request by zshainsky**
Please review the Objects and Libraries definition page

### CODE REVIEW

**Comment by zshainsky** [comment]
I am no Git expert... so my SAML updates got added in this as well :/

**Comment by zshainsky** [comment]
"--help" commit is my commit with OIDC update for Okta tile

File: docs/guide/datasource/datasource_objects.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+| R / SparkR | `conn <- z.getDatasource("<data source name>")` | [DBIConnection object:](https://www.rdocumentation.org/packages/DBI/versions/0.5-1/topics/DBIConnection-class) Zepl function uses the `dbConnect()` function from the DPLYR library |  |
+| Scala | ```val df = z.getDatasource("<data source name>").asInstanceOf[org.apache.spark.sql.DataFrameReader]``` | Spark SQL Dataframe: [org.apache.spark.sql.DataFrameReader](https://spark.apache.org/docs/2.3.1/api/java/org/apache/spark/sql/DataFrameReader.html) |
+| SQL | `%datasource.<data source name>` | JDBC connection to Snowflake | |
+
+>Note: Spark (Scala and Pyspark) require 'STAGE' permissions on the Snowflake database or else this exception may be thrown: `net.snowflake.client.jdbc.SnowflakeSQLException: SQL execution error: Creating stage on shared database 'SNOWFLAKE_SAMPLE_DATA' is not allowed`
+
+### Write data to S3
```
Comment (louis-zepl) [inline docs/guide/datasource/datasource_objects.md]: This should be Write Data to Snowflake

**Review by louis-zepl — COMMENTED** [review]
Just found a minor typo

File: docs/guide/datasource/datasource_objects.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
+# Data Source Objects and Libraries
+This article will cover all of the data types that are returned by Zepl’s secure data source objects. In order to build a standard method for accessing any data source, Zepl has built data source connectors to support one or all of the four programming languages used by data scientists; Python, R, Scala, and SQL.
+
+Zepl has created a simple, secure, and standard method for all users to access their data. Zepl’s data sources allow each user to securely store their user credentials through our encrypted keychain. 
+
+Each programming language supports different access methods to each of these data sources. We will walk through what objects are return upon connecting to each data source. This will give you the best insight to take advantage of our features and troubleshoot anytime errors may arise.
```
Comment (BryanKatis) [inline docs/guide/datasource/datasource_objects.md]: 1. change 'return' to 'returned'
2. change 'troubleshoot anytime errors may arise.' to 'troubleshoot any errors that may arise.'

File: docs/guide/datasource/datasource_objects.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+* [Snowflake](#snowflake)
+* [S3](#s3)
+* [Google BigQuery](#google-bigquery)
+* [MySQL](#mysql)
+* [PosgreSQL](#posgresql)
```
Comment (BryanKatis) [inline docs/guide/datasource/datasource_objects.md]: Should be 'PostgreSQL', 'Apache Cassandra', and 'Amazon S3'

File: docs/guide/datasource/datasource_objects.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+### Write data to MySQL
+>Example coming soon!
+
+# PosgreSQL
+### Configure
+* Host (required)
+* Port (required)
+* Database (optional)
+
+### Read data from PosgreSQL
```
Comment (BryanKatis) [inline docs/guide/datasource/datasource_objects.md]: Should be 'PostgreSQL'

**Review by BryanKatis — APPROVED** [review]

**Pull request merged**